cpp/tests/strings/array_tests.cu

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/sorting.hpp>
+#include <cudf/strings/copying.hpp>
+#include <utilities/error_utils.hpp>
+
+#include <tests/utilities/base_fixture.hpp>
+#include "./utilities.h"
+
+#include <vector>
+#include <gmock/gmock.h>
+
+
+struct StringsColumnTest : public cudf::test::BaseFixture {};
+
+TEST_F(StringsColumnTest, Sort)
+{
+    std::vector<const char*> h_strings{ "eee", "bb", nullptr, "", "aa", "bbb", "ééé" };
+    std::vector<const char*> h_expected{ nullptr, "", "aa", "bb", "bbb", "eee", "ééé" };
+
+    auto d_strings = cudf::test::create_strings_column(h_strings);
+    auto strings_view = cudf::strings_column_view(d_strings->view());
+
+    auto results = cudf::strings::detail::sort(strings_view, cudf::strings::detail::name);
+    cudf::test::expect_strings_equal(*results, h_expected);
+}
+
+TEST_F(StringsColumnTest, SortZeroSizeStringsColumn)
+{
+    cudf::column_view zero_size_strings_column( cudf::data_type{cudf::STRING}, 0, nullptr, nullptr, 0);
+    auto strings_view = cudf::strings_column_view(zero_size_strings_column);
+    EXPECT_THROW( cudf::strings::detail::sort(strings_view, cudf::strings::detail::name), cudf::logic_error );
```

Comment:
I don't think zero-size columns should throw errors. Dask expects empty inputs to produce empty outputs.